src/components/BasicTableModal/index.jsx

```diff
@@ -9,20 +10,136 @@ import {
+import saveEditedQuantity from '../../service/save-quantity';
+import useSWR from 'swr';
+import { fetcher, stockRoomURL, stockTakeURL } from '../../utils/api-utils';
+import { locationCookieName } from '../../../constants';
+import { useCookies } from 'react-cookie';
 
 const TableModal = (props) => {
-  const { showModal, headers, rows, closeModal, stickyColumnName, title } = props;
+  const {
+    showModal,
+    headers,
+    rows,
+    closeModal,
+    stickyColumnName,
+    title,
+    onChildSave,
+
+  } = props;
+
+  const [editedRows, setEditedRows] = useState([...rows]);
+  const [editingRowId, setEditingRowId] = useState(null);
+  const [cookies] = useCookies();
+  const [saveClicked, setSaveClicked] = useState(false);
+  const [isEditing, setIsEditing] = useState(false); // Track editing mode
+  const editRowRef = useRef(null);
+  const [editedQuantity, setEditedQuantity] = useState({}); // State to track edited quantity values
+
+  const { data: stockRoom, error: stockRoomError } = useSWR(
+    stockRoomURL(cookies[locationCookieName]?.name.trim()),
+    fetcher,
+  );
+
+  useEffect(() => {
+    if (saveClicked) {
+      onChildSave();
+      setSaveClicked(false);
+    }
+  }, [saveClicked, onChildSave]);
+
+useEffect(() => {
+    setEditedRows([...rows]);
+  }, [rows]);
+  useEffect(() => {
+    function handleClickOutside(event) {
+      if (editRowRef.current && !editRowRef.current.contains(event.target)) {
+        setIsEditing(false);
+        setEditingRowId(null);
+      }
+    }
+
+    document.addEventListener('mousedown', handleClickOutside);
+
+    return () => {
+      document.removeEventListener('mousedown', handleClickOutside);
+    };
+  }, []);
+
+  const handleQuantityChange = (rowId, value) => {
+    // Update the edited quantity state when the quantity input changes
+    setEditedQuantity((prevEditedQuantity) => ({
+      ...prevEditedQuantity,
+      [rowId]: value,
+    }));
+  };
+
+  const handleActualQuantityChange = (rowId, value) => {
+    const updatedEditedRows = editedRows.map((row) => {
+      if (row.id === rowId) {
+        return {
+          ...row,
+          actualQuantity: value,
+        };
+      }
+      return row;
+    });
+    setEditedRows(updatedEditedRows);
+  };
+
+  const startEditingRow = (rowId) => {
+    setEditingRowId(rowId);
+
+    setIsEditing(true); // Enter editing mode
+  };
+
+  const handleSaveClick = async (rowId) => {
+    try {
+      const editedRow = editedRows.find((row) => row.id === rowId);
+
+      if (editedRow) {
+        const { productName, quantity, actualQuantity, expiration, batchNumber } = editedRow;
+
+        // Use the edited quantity state if available, otherwise use the original quantity value
+        const editedQuantityValue = editedQuantity[rowId] || quantity;
+
+        const response = await saveEditedQuantity(
+          productName,
+          editedQuantityValue, // Use the edited quantity value
```

Comment:
Why is `editedQuantityValue` passed over here? How is it different from `actualQuantity`

Reply:
Sure this was added as part of earlier implementation and is unused now removed it.